students/HABTAMU/lesson06/assignment/good_perf.py

```diff
@@ -0,0 +1,37 @@
+"""
+poorly performing, poorly written module
+"""
+
+import datetime
+import csv
+from collections import defaultdict
+
+def analyze(filename):
+    start = datetime.datetime.now()
+    year_count = defaultdict(lambda : 0)
+
+    with open(filename) as csvfile:
+        reader = csv.reader(csvfile, delimiter=',', quotechar='"')
+        new_ones = []
+        found = 0
+        for row in reader:
+            if (row[5][6:] > '2012') and (row[5][6:] < '2019'):
+                year_count[row[5][6:]] += 1
+
+            if "ao" in row[6]:
+                found +=1
+
+        year_count = dict(year_count)
```

Comment:
You don't need this line, and it probably is slowing you down a bit. Defaultdict is a dictionary, even though it has the weird signature.

Reply:
the reason i added that was to remove that weird object text ... 
this is what am talking about... `defaultdict(<function analyze.<locals>.<lambda> at 0x109d3fa60>, {'2015': 5994, '2013': 5911, '2014': 5854, '2016': 5762, '2017': 5789, '2018': 5811})` when i use dict(year) it will not show that again.